### PR TITLE
merge 2.2 (DlgTrackInfo)

### PR DESCRIPTION
**Pull request by daschuer**
This is basically a merge of the 2.2 branch into master. 
Unfortunately It does not work clean so I like to have a second pair of eyes on this before merging into master.

### CODE REVIEW

**Comment by uklotzde** [comment]
Test failures?

**Comment by daschuer** [comment]
Ups .. fixed.

File: src/library/dlgtrackinfo.cpp [hunk 1]
```diff
@@ -423,23 +443,25 @@ void DlgTrackInfo::saveTrack() {
...
         }
 
-        auto colorComboBox = qobject_cast<QComboBox*>(colorWidget);
-        if (colorComboBox) {
-            PredefinedColorPointer color = Color::kPredefinedColorsSet.allColors.at(colorComboBox->currentIndex());
-            pCue->setColor(color);
+        if (pCue->getType() == Cue::CUE) {
+            auto colorComboBox = qobject_cast<QComboBox*>(colorWidget);
+            if (colorComboBox) {
+                PredefinedColorPointer color = Color::kPredefinedColorsSet.allColors.at(colorComboBox->currentIndex());
```
Comment (uklotzde) [inline src/library/dlgtrackinfo.cpp]: Just for the record: Obtaining a value and passing it along is a typical use case where `auto` is justified. Repeating the actual type if you don't even use it adds nothing.

**Comment by uklotzde** [comment]
Thank you, Daniel! Waiting for the macOS build to succeed, just to be sure.

**Pull request merged**